applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -526,7 +525,8 @@ def GetShapeGradient(self):
 
 
     def FinalizeSolutionStep(self):
-        self.adjoint_analysis.FinalizeSolutionStep()
+        # Commented out because this causes finalize solution step to be called twice
+        #self.adjoint_analysis.FinalizeSolutionStep()
```

Comment:
i think you can even remove the `OutputSolutionStep` and therefor completely remove this function from your python response - look at the AdjointResponse for the linear case.

Maybe just double check that also this output `OutputSolutionStep` is already called in the "RunSolutionLoop" function of the analysis.

Reply:
@armingeiser Okay, I will check it. right now this fixes the issue we discussed today that was caused by calling UpdateSensitivities() in FinalizeSolutionStep()